### PR TITLE
ci: use latest action to replicate workflows to other repos

### PR DESCRIPTION
**Pull request by derberg**
**Description**

- use new version of action that copies files from this repo to other repos. It now supports not only workflow files
- this change do not handle cleanup. Cleanup like `removal go-related workflows from node project`, or `nodejs-related workflows from go projects` is a manual work that has to be done by owners of related repositories when they have time. This action and workflow do not detect files that are no longer needed and therefor should be removed from some repo.
- I don't think we should replicate other files than contributing guide and code of conduct. Imho support and other files can stay here as they are anyway supported by GitHub properly

TODO:
- before it is merged I need to make sure `CONTRIBUTING.md` is only different for spec and community. Maybe other repos have their own versions
- before it is merged I need to add proper topics to Go and NodeJS projects

**Related issue(s)**
Resolves https://github.com/asyncapi/.github/issues/39
Resolves https://github.com/asyncapi/.github/issues/40

@jonaslagoni pinging you as you are the owner of related issues
@smoya pinging you as you are owner of go projects
@KhudaDad414 pinging you for obvious reasons - I need your review
@magicmatatjahu pinging you so you know we can also have docker only workflows distributed only to repost with `docker` topic. Also new action also supports `branches` option, so we can do replication of files in branches different than main, in case we want.

### CODE REVIEW

**Review by KhudaDad414 — APPROVED** [review]
Thanks, @derberg 🙇. love the new name of the action :)

**Comment by KhudaDad414** [comment]
@HarshCasper you may be interested in this PR as well, for https://github.com/asyncapi/.github/pull/147

**Comment by derberg** [comment]
@magicmatatjahu @jonaslagoni I went through all repos that have their own contributing guide. Some of them I ignore in new workflows. I need your opinion on what about:

- https://github.com/asyncapi/asyncapi-react/blob/master/CONTRIBUTING.md
- https://github.com/asyncapi/modelina/blob/master/docs/contributing.md
- https://github.com/asyncapi/dotnet-nats-template/blob/master/CONTRIBUTING.md
- https://github.com/asyncapi/ts-nats-template/blob/master/CONTRIBUTING.md

my suggestion is to take "custom" parts into README.md into `Development` section, or `Contributing` section + switch to generic `CONTRIBUTING.md`

**Comment by magicmatatjahu** [comment]
@derberg To correctly understand: do we need to remove our custom contribution guide and link to the `CONTRIBUTING.md` which will be copied to the every repo by "global" workflow or should this link point to a contribution in the community repo?

**Comment by derberg** [comment]
@magicmatatjahu you can have your custom CONTRIBUTING guide as you have now, but you can also have the one that we will copy over. Up to you. You will just need to move "custom" parts somewhere so they are not lost

File: .github/workflows/global-replicator.yml [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+      runs-on: ubuntu-latest
+      steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      - name: Replicating file
+        uses: derberg/copy-files-to-other-repositories@v1.0.0
+        with:
+          github_token: ${{ secrets.GH_TOKEN }}
+          patterns_to_include: .github/workflows/if-nodejs-pr-testing.yml,.github/workflows/if-nodejs-release.yml,.github/workflows/if-nodejs-version-bump.yml,.github/workflows/bump.yml
+          topics_to_include: nodejs
```
Comment (smoya) [inline .github/workflows/global-replicator.yml:79]: `topics_to_include` is key. This is awesome. This is amazing 🥇 💯 👏

File: .github/workflows/global-replicator.yml [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    branches: [ master ]
+    paths:
+      - '.github/workflows/**'
+      - 'CODE_OF_CONDUCT.md'
+      - 'CONTRIBUTING.md'
+  workflow_dispatch:
+    inputs:
+      repo_name:
+        description: |
+          You can specify name of the repository where workflows should be pushed manually. As long as repository is not ignored by workflow settings.
```
Comment (alequetzalli) [inline .github/workflows/global-replicator.yml:14]:
```suggestion
          You can specify the repository's name where workflows should be pushed manually, as long as workflow settings do not ignore the repository.
```

File: .github/workflows/global-replicator.yml [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+          If you do not specify exact repository name, the workflow will try to replicate all missing changes to all repositories.
```
Comment (alequetzalli) [inline .github/workflows/global-replicator.yml:15]:
```suggestion
          If you do not specify the exact repository name, the workflow will try to replicate all missing changes to all repositories.
```

**Review by alequetzalli — COMMENTED** [review]
Just 2 tiny items 🎉🎉🎉

**Review by alequetzalli — DISMISSED** [review]
🚢 🚢 🚢

**Comment by derberg** [comment]
@magicmatatjahu @jonaslagoni so what is your take on mentioned CONTRIBUTING guides? should I add them to `ignored` for now?

**Comment by jonaslagoni** [comment]
> @magicmatatjahu @jonaslagoni so what is your take on mentioned CONTRIBUTING guides? should I add them to `ignored` for now?

Sounds good 👍

**Comment by derberg** [comment]
@magicmatatjahu pingy pongy

**Comment by magicmatatjahu** [comment]
@derberg Sorry, I missed that notification. You can go ahead, as I see with `react-component` won't be problem.

**Comment by derberg** [comment]
all adjusted as requested
we also have `nodejs` topic added where needed, same with `go` thanks to @smoya 

I need just one 🍏

**Review by smoya — DISMISSED** [review]
🚀 🌔

**Review by magicmatatjahu — APPROVED** [review]
LGTM!

**Review by KhudaDad414 — APPROVED** [review]
LGTM!

**Review by alequetzalli — APPROVED** [review]

**Pull request merged**